libchisel/src/verifyimports.rs

```diff
@@ -0,0 +1,389 @@
+use super::ModuleValidator;
+
+use parity_wasm::elements::{
+    External, FunctionType, ImportEntry, ImportSection, Module, Type, ValueType,
+};
+
+/// Enum representing a type of import and any extra data to check.
+#[derive(Clone)]
+pub enum ImportType<'a> {
+    Function(&'a str, &'a str, FunctionType),
+    Global(&'a str, &'a str),
+    Memory(&'a str, &'a str),
+    Table(&'a str, &'a str),
+}
+
+/// Trait over ImportType that lets a caller check if it is imported in a given module.
+trait IsImported {
+    fn is_imported(&self, module: &Module) -> bool;
+}
+
+/// Struct on which ModuleValidator is implemented.
+pub struct VerifyImports<'a> {
+    entries: Vec<ImportType<'a>>,
+    require_all: bool,
+    allow_unlisted: bool,
+}
+
+impl<'a> VerifyImports<'a> {
+    fn ewasm() -> Self {
+        VerifyImports {
+            //FIXME: Do not inline this
+            entries: vec![
+                ImportType::Function(
+                    "ethereum",
+                    "useGas",
+                    FunctionType::new(vec![ValueType::I64], None),
+                ),
+                ImportType::Function(
+                    "ethereum",
+                    "getGasLeft",
+                    FunctionType::new(Vec::new(), Some(ValueType::I64)),
```

Comment:
Cannot you use `vec![]` here?

Reply:
yeah. not much of a difference, I would say.

Reply:
Though it is more consistent with the other inline declaration of `vec`.

Reply:
true, will fix

Reply:
fixed.